activestorage/lib/active_storage/attached/macros.rb

```diff
@@ -83,7 +83,21 @@ def #{name}=(attachables)
         end
       CODE
 
-      has_many :"#{name}_attachments", -> { where(name: name) }, as: :record, class_name: "ActiveStorage::Attachment", inverse_of: :record
+      has_many :"#{name}_attachments", -> { where(name: name) }, as: :record, class_name: "ActiveStorage::Attachment", inverse_of: :record do
+        def purge
+          if any?
+            each(&:purge)
+            reset
+          end
+        end
+
+        def purge_later
+          if any?
+            each(&:purge_later)
```

Comment:
I think that we can do `each(&:purge_later)` directly without `if any?`.
`if any?` may (almost) causes extra query.

Reply:
If I take it off both and then test `assert_nothing_raised { @user.highlights.purge }` and `purge_later` without creating any blobs, no error is raised. Same if I add a lookup for an id that doesn't exist. What is the `if any?` catching?

Reply:
@nicholasshirley I don't understand your comment. Can you elaborate.

> What is the `if any?` catching?

I'm not sure, if @kamipo is doubting that we need it, we can surely remove it.